StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1615UnitTests.cs

```diff
@@ -106,14 +133,106 @@ public class ClassName
     /// </summary>
 $$
 }";
-            var expected = new[]
-            {
-                this.CSharpDiagnostic().WithLocation(10, 21)
-            };
+            var fixedCode = @"
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+    /// <returns></returns>
+$$
+}";
+
+            var expected = this.CSharpDiagnostic().WithLocation(10, 21);
 
             await this.VerifyCSharpDiagnosticAsync(testCode.Replace("$$", declaration), expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode.Replace("$$", declaration), EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode.Replace("$$", declaration), fixedCode.Replace("$$", declaration), cancellationToken: CancellationToken.None).ConfigureAwait(false);
         }
-        
+
+        [Theory]
+        [MemberData(nameof(AsynchronousWithReturnValue))]
+        public async Task TestAsynchronousMethodWithReturnTypeWithoutReturnTypeDocumentationAsync(string declaration)
+        {
+            var testCode = @"
+using System.Threading.Tasks;
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+$$
+}";
+            var fixedCode = @"
+using System.Threading.Tasks;
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+    /// <returns><placeholder>A <see cref=""Task""/> representing the asynchronous operation.</placeholder></returns>
```

Comment:
Ah, I see thanks.
Did you really mean *anytime* doc is generated by an analyzer? That seems like it would apply to the boilerplate doc produced for the `<summary>` tag of a constructor, and that bolierplate is adequate IMO as-is, and adding `<placeholder>` around it every time the code fixer spits it out would make it more work for the user every time it was produced.